vmm/src/lib.rs

```diff
@@ -2035,24 +2177,38 @@ impl RequestHandler for Vmm {
                         continue;
                     }
 
-                    let mut buf = [0u8; 4];
-                    let (_, file) = socket.recv_with_fd(&mut buf).map_err(|e| {
-                        MigratableError::MigrateReceive(anyhow!(
-                            "Error receiving slot from socket: {}",
-                            e
-                        ))
-                    })?;
+                    match &mut socket {
+                        SocketStream::Unix(unix_socket) => {
+                            let mut buf = [0u8; 4];
+                            let (_, file) = unix_socket.recv_with_fd(&mut buf).map_err(|e| {
+                                MigratableError::MigrateReceive(anyhow!(
+                                    "Error receiving slot from socket: {}",
+                                    e
+                                ))
+                            })?;
+
+                            if existing_memory_files.is_none() {
+                                existing_memory_files = Some(HashMap::default())
+                            }
 
-                    if existing_memory_files.is_none() {
-                        existing_memory_files = Some(HashMap::default())
-                    }
+                            if let Some(ref mut existing_memory_files) = existing_memory_files {
+                                let slot = u32::from_le_bytes(buf);
+                                existing_memory_files.insert(slot, file.unwrap());
+                            }
 
-                    if let Some(ref mut existing_memory_files) = existing_memory_files {
-                        let slot = u32::from_le_bytes(buf);
-                        existing_memory_files.insert(slot, file.unwrap());
+                            Response::ok().write_to(&mut socket)?;
+                        }
+                        SocketStream::Tcp(_tcp_socket) => {
+                            // For TCP sockets, we cannot transfer file descriptors
+                            warn!(
+                                "MemoryFd command received over TCP socket, which is not supported"
+                            );
+                            Response::error().write_to(&mut socket)?;
+                            return Err(MigratableError::MigrateReceive(anyhow!(
+                                "MemoryFd command not supported over TCP sockets"
+                            )));
```

Comment:
Let's drop the `return Err()`.

I don't think we will need to abort the destination VM explicitly here, since it is mostly driven by the source VM. Let's leave it up to the source VM to decide what to do when there is an error returned from the destination VM side. As for now, it will send the "Abandon" request to stop the destination VM.

Reply:
> Let's drop the `return Err()`.
> 
> I don't think we will need to abort the destination VM explicitly here, since it is mostly driven by the source VM. Let's leave it up to the source VM to decide what to do when there is an error returned from the destination VM side. As for now, it will send the "Abandon" request to stop the destination VM.

I have drop the `return Err()`, thanks.